crates/forge/src/lib.rs

```diff
@@ -349,6 +354,8 @@ async fn run_tests_from_crate(
     drop(send_shut_down);
     let _ = rec_shut_down.recv().await;
 
+    assert!(!interrupted, "Tests were interrupted");
```

Comment:
I think a comment explaining the logic behind this assert could be useful here. It's not 100% obvious why we want to panic if this is reached.